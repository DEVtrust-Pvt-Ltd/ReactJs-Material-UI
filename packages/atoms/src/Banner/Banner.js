import React, { useCallback } from "react";
import PropTypes from "prop-types";
import { Box, makeStyles, useTheme } from "@material-ui/core";
import SwipeableViews from "react-swipeable-views";
import { autoPlay } from "react-swipeable-views-utils";

const useStyles = makeStyles((theme) => ({
  banner: {
    height: "auto",
    display: "block",
    overflow: "hidden",
    width: "100%",
    marginBottom: theme.spacing(8),
    paddingTop: theme.spacing(0),
    cursor: "pointer",
  },
  iconsPosition: {
    position: "relative",
  },
}));

const AutoPlaySwipeableViews = autoPlay(SwipeableViews);

export const Banner = ({ banners, onBannerClick }) => {
  const classes = useStyles();
  const theme = useTheme();
  const [activeStep, setActiveStep] = React.useState(1);
  const handleStepChange = useCallback((step) => setActiveStep(step), [
    activeStep,
  ]);

  return (
    <Box className={classes.iconsPosition}>
      <AutoPlaySwipeableViews
        data-testid="auto-play-swipeable-views"
        axis={theme.direction === "rtl" ? "x-reverse" : "x"}
        index={activeStep}
        onChangeIndex={handleStepChange}
        enableMouseEvents
      >
        {banners.map((banner, index) => (
          <Box key={banner.id}>
            {Math.abs(activeStep - index) <= banners.length ? (
              <img
                className={classes.banner}
                src={banner.imageUrl}
                alt="Images"
                onClick={onBannerClick}
                role="presentation"
              />
            ) : null}
          </Box>
        ))}
      </AutoPlaySwipeableViews>
    </Box>
  );
};

Banner.propTypes = {
  banners: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.shape({})),
  ]),
  secureBaseMediaUrl: PropTypes.oneOfType([PropTypes.string, PropTypes.string]),
  onBannerClick: PropTypes.func.isRequired,
};

Banner.default = {
  banners: [],
  secureBaseMediaUrl: null,
};

export default Banner;
