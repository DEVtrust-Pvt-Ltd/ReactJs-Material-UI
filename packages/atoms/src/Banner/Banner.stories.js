import React from "react";
import { storiesOf } from "@storybook/react";
import { withKnobs, object } from "@storybook/addon-knobs";
import { action } from '@storybook/addon-actions'

import Banner from ".";

const banners = [
  {
    id: 1,
    imageUrl: "https://material-ui.com/static/images/cards/paella.jpg",
  },
  {
    id: 2,
    imageUrl:
      "https://material-ui.com/static/images/cards/contemplative-reptile.jpg",
  },
];

storiesOf("Atoms|Banners", module)
  .addDecorator(withKnobs)
  .add("banner", () => (
    <Banner
      banners={object("banners (prop)", banners)}
      onBannerClick={action('onBannerClick')}
    />
  ));
