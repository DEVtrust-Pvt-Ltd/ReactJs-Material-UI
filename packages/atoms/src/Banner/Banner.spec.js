import React from "react";
import {fireEvent, render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Banner from "./Banner";

const bannerDetail = [
    {
        id: 1,
        image:
            "pub/media/weltpixel/owlcarouselslider/images/j/u/cache/1200x350/july_cancer_prevention.png",
    },
    {
        id: 2,
        image:
            "pub/media/weltpixel/owlcarouselslider/images/l/a/cache/1200x350/landing_page-02_1_.png",
    },
];

const secureBaseMediaUrl = "https://choices.cxapalawan.com";

const handleBannerClick = () => {
    alert("banner clicked....");
};

describe("Banner Component", () => {
    it("should match snapshot when properties are present", () => {
        const {container} = render(
            <Banner
                banners={bannerDetail}
                secureBaseMediaUrl={secureBaseMediaUrl}
                onBannerClick={handleBannerClick}
            />
        );

        expect(container).toMatchSnapshot();
    });

    it("should call event", () => {
        const Component = Banner;

        const {getByTestId} = render(
            <Component
                banners={bannerDetail}
                secureBaseMediaUrl={secureBaseMediaUrl}
                onBannerClick={handleBannerClick}
            />
        );
        const swipableView = getByTestId("auto-play-swipeable-views");
        fireEvent.change(swipableView)

    });

});
