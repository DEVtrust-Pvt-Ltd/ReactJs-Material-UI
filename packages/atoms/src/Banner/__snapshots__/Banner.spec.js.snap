// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Component should match snapshot when properties are present 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-iconsPosition-2"
  >
    <div
      data-testid="auto-play-swipeable-views"
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="MuiBox-root MuiBox-root-5"
          >
            <img
              alt="Images"
              class="makeStyles-banner-1"
              role="presentation"
              src="https://choices.cxapalawan.com/pub/media/weltpixel/owlcarouselslider/images/l/a/cache/1200x350/landing_page-02_1_.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
